code/core/src/androidTest/java/com/adobe/marketing/mobile/internal/DataMarshallerTests.kt

```diff
@@ -206,6 +209,19 @@ class DataMarshallerTests {
             result
         )
     }
+
+    @Test
+    fun marshalIntentExtras_whenBundleKeySetThrowsException_NoCrash() {
+
+        val mockIntent = Mockito.mock(Intent::class.java)
+        val mockBundle = Mockito.mock(Bundle::class.java)
+
+        `when`(mockIntent.extras).thenReturn(mockBundle)
+        `when`(mockBundle.keySet()).thenThrow(RuntimeException::class.java)
```

Comment:
do we have any options to mimic a bad parcel write to get closer to the real behavior?

Reply:
Nope